kategory/src/main/kotlin/kategory/instances/NonEmptyListInstances.kt

```diff
@@ -32,6 +34,23 @@ interface NonEmptyListInstances :
         }
     }
 
+    override fun <A, B> foldL(fa: HK<NonEmptyListHK, A>, b: B, f: (B, A) -> B): B = fa.ev().tail.fold(f(b, fa.ev().head), f)
+
+    override fun <A, B> foldR(fa: HK<NonEmptyListHK, A>, lb: Eval<B>, f: (A, Eval<B>) -> Eval<B>): Eval<B> {
+        fun loop(fa_p: NonEmptyList<A>): Eval<B> = when {
```

Comment:
Yes, I mean, is ListKW.foldR stack-safe :D That's beyond the scope of this ticket. I'll approve as it is, we can investigate later.